### ISSUE TITLE
v1.1.0-beta1 [Syntax Error] line 0, col 85: Error: Expected Literal, got 'order'"

### ISSUE THREAD

**Issue by jdeniau** [dunglas/DunglasApiBundle#341]
I got an `Invoice` entity with an `order` link on an `Order` Entity.

I can filter the invoices by order (`/invoices/?order=/order/8`)

I tried updating to the v1.1.0-beta1 but then I got this error:

```json
{
  "@context": "/v1/contexts/Error",
  "@type": "Error",
  "hydra:title": "An error occurred",
  "hydra:description": "[Syntax Error] line 0, col 85: Error: Expected Literal, got 'order'",
  "trace": [
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 448,
      "function": "syntaxError",
      "class": "Doctrine\\ORM\\Query\\QueryException",
      "type": "::",
      "args": [
        "line 0, col 85: Error: Expected Literal, got 'order'",
        {

        }
      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2586,
      "function": "syntaxError",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [
        "Literal"
      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2774,
      "function": "Literal",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2708,
      "function": "ArithmeticPrimary",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2676,
      "function": "ArithmeticFactor",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2650,
      "function": "ArithmeticTerm",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2637,
      "function": "SimpleArithmeticExpression",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2973,
      "function": "ArithmeticExpression",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2501,
      "function": "ComparisonExpression",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2384,
      "function": "SimpleConditionalExpression",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2360,
      "function": "ConditionalPrimary",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2328,
      "function": "ConditionalFactor",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 2303,
      "function": "ConditionalTerm",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 1327,
      "function": "ConditionalExpression",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 876,
      "function": "WhereClause",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 843,
      "function": "SelectStatement",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 268,
      "function": "QueryLanguage",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php",
      "line": 351,
      "function": "getAST",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Tools/Pagination/Paginator.php",
      "line": 260,
      "function": "parse",
      "class": "Doctrine\\ORM\\Query\\Parser",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/doctrine/orm/lib/Doctrine/ORM/Tools/Pagination/Paginator.php",
      "line": 123,
      "function": "getCountQuery",
      "class": "Doctrine\\ORM\\Tools\\Pagination\\Paginator",
      "type": "->",
      "args": [

      ]
    },
    {
      "function": "count",
      "class": "Doctrine\\ORM\\Tools\\Pagination\\Paginator",
      "type": "->",
      "args": [

      ]
    },
    {
      "file": "vendor/dunglas/api-bundle/Doctrine/Orm/Paginator.php",
      "line": 51,
      "function": "count",
      "args": [
        {

        }
      ]
    },
    {
      "file": "vendor/dunglas/api-bundle/Doctrine/Orm/DataProvider.php",
      "line": 156,
      "function": "__construct",
      "class": "Dunglas\\ApiBundle\\Doctrine\\Orm\\Paginator",
      "type": "->",
      "args": [
        {

        }
      ]
    },
    {
      "file": "vendor/dunglas/api-bundle/Doctrine/Orm/DataProvider.php",
      "line": 140,
      "function": "getPaginator",
      "class": "Dunglas\\ApiBundle\\Doctrine\\Orm\\DataProvider",
      "type": "->",
      "args": [
        {

        }
      ]
    },
    {
      "file": "vendor/dunglas/api-bundle/Model/DataProviderChain.php",
      "line": 55,
      "function": "getCollection",
      "class": "Dunglas\\ApiBundle\\Doctrine\\Orm\\DataProvider",
      "type": "->",
      "args": [
        {

        },
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        }
      ]
    },
    {
      "file": "app/cache/dev/appDevDebugProjectContainer.php",
      "line": 8722,
      "function": "getCollection",
      "class": "Dunglas\\ApiBundle\\Model\\DataProviderChain",
      "type": "->",
      "args": [
        {

        },
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        }
      ]
    },
    {
      "file": "vendor/dunglas/api-bundle/Controller/ResourceController.php",
      "line": 135,
      "function": "getCollection",
      "class": "DunglasApiBundleModelDataProviderChain_00000000135e863e0000735398f837533b78331467f762743773e8eef13aa056",
      "type": "->",
      "args": [
        {

        },
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        }
      ]
    },
    {
      "file": "vendor/dunglas/api-bundle/Controller/ResourceController.php",
      "line": 150,
      "function": "getCollectionData",
      "class": "Dunglas\\ApiBundle\\Controller\\ResourceController",
      "type": "->",
      "args": [
        {

        },
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        }
      ]
    },
    {
      "function": "cgetAction",
      "class": "Dunglas\\ApiBundle\\Controller\\ResourceController",
      "type": "->",
      "args": [
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        }
      ]
    },
    {
      "file": "app/bootstrap.php.cache",
      "line": 3109,
      "function": "call_user_func_array",
      "args": [
        [
          {

          },
          "cgetAction"
        ],
        [
          {
            "attributes": {

            },
            "request": {

            },
            "query": {

            },
            "server": {

            },
            "files": {

            },
            "cookies": {

            },
            "headers": {

            }
          }
        ]
      ]
    },
    {
      "file": "app/bootstrap.php.cache",
      "line": 3071,
      "function": "handleRaw",
      "class": "Symfony\\Component\\HttpKernel\\HttpKernel",
      "type": "->",
      "args": [
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        },
        1
      ]
    },
    {
      "file": "app/bootstrap.php.cache",
      "line": 3222,
      "function": "handle",
      "class": "Symfony\\Component\\HttpKernel\\HttpKernel",
      "type": "->",
      "args": [
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        },
        1,
        true
      ]
    },
    {
      "file": "app/bootstrap.php.cache",
      "line": 2444,
      "function": "handle",
      "class": "Symfony\\Component\\HttpKernel\\DependencyInjection\\ContainerAwareHttpKernel",
      "type": "->",
      "args": [
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        },
        1,
        true
      ]
    },
    {
      "file": "web/app_dev.php",
      "line": 31,
      "function": "handle",
      "class": "Symfony\\Component\\HttpKernel\\Kernel",
      "type": "->",
      "args": [
        {
          "attributes": {

          },
          "request": {

          },
          "query": {

          },
          "server": {

          },
          "files": {

          },
          "cookies": {

          },
          "headers": {

          }
        }
      ]
    }
  ]
}

```

**Comment by vincentchalamon** [comment]
Hi,

I think it's a conflict against the `order` query parameter name. Did you try to [use a custom order query parameter name](https://api-platform.com/doc/1.0/api-bundle/filters#using-a-custom-order-query-parameter-name) ?

**Comment by theofidry** [comment]
@vincentchalamon I though invalid values for filters are simply ignored?

**Comment by vincentchalamon** [comment]
@theofidry I also thought: https://github.com/dunglas/DunglasApiBundle/blob/master/Doctrine/Orm/Filter/OrderFilter.php#L84

**Comment by theofidry** [comment]
At some point we'll definitely have to address this log/debug issue cf. #102

**Comment by jdeniau** [comment]
@vincentchalamon yes, I tried, but no changes

**Comment by theofidry** [comment]
@jdeniau Did you try to dump the query and its parameters?

**Comment by teohhanhui** [comment]
@jdeniau I think it must be something in `DataProvider::useOutputWalkers`.

But honestly, it's a lot of overhead to support reserved words which shouldn't be used in the first place. At least I'd argue that if it's not supported by Doctrine, I don't see why we should...

**Comment by theofidry** [comment]
@teohhanhui you mean the order filter?

**Comment by dunglas** [comment]
Agree with @teohhanhui it's out of scope of this project, it's the responsibility of the persistence layer or of the developer.

**Comment by teohhanhui** [comment]
@theofidry Looking again at the above stack trace (even though the line numbers don't match), it looks like I was wrong. This exception is thrown during the construction of the Doctrine `Paginator` so the `QueryBuilder` must be using the reserved word...

**Comment by teohhanhui** [comment]
And no @theofidry this exception has nothing to do with the order filter. If you look at https://github.com/dunglas/DunglasApiBundle/blob/v1.1.0-beta.1/Doctrine/Orm/Filter/OrderFilter.php#L56-58 it's clear that it doesn't do anything.

**Comment by jdeniau** [comment]
Well I don't think I understand all as you have a better visibility on the project than me, but the only think I got here is a ManyToOne relation with an attribute named `$order`. 

It works perfectly fine in doctrine, and it works perfectly fine in api-bundle v1.0. 

If I named my attribute differently (without changing anything else in doctrine or in my database), it works fine.
I will try to investigate further though to find the breaking point of this.
